src/source/Response.c

```diff
@@ -445,24 +469,24 @@ STATUS curlCompleteSync(PCurlResponse pCurlResponse)
 STATUS notifyDataAvailable(PCurlResponse pCurlResponse, UINT64 durationAvailable, UINT64 sizeAvailable)
 {
     STATUS retStatus = STATUS_SUCCESS;
-    CURLcode result;
+    PCallbacksProvider pCallbacksProvider;
 
     CHK(pCurlResponse != NULL, STATUS_NULL_ARG);
 
+    if (pCurlResponse->pCurlRequest != NULL && pCurlResponse->pCurlRequest->pCurlApiCallbacks != NULL &&
+        pCurlResponse->pCurlRequest->pCurlApiCallbacks->pCallbacksProvider != NULL) {
+        pCallbacksProvider = pCurlResponse->pCurlRequest->pCurlApiCallbacks->pCallbacksProvider;
+    }
+
     // pCurlResponse should be a putMedia session
     if (!ATOMIC_LOAD_BOOL(&pCurlResponse->terminated)) {
         DLOGV("Note data received: duration(100ns): %" PRIu64 " bytes %" PRIu64 " for stream handle %" PRIu64, durationAvailable, sizeAvailable,
               pCurlResponse->pCurlRequest->uploadHandle);
 
-        if (pCurlResponse->paused && pCurlResponse->pCurl != NULL) {
-            pCurlResponse->paused = FALSE;
-            // frequent pause unpause causes curl segfault in offline scenario
-            THREAD_SLEEP(10 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
-            // un-pause curl
-            result = curl_easy_pause(pCurlResponse->pCurl, CURLPAUSE_SEND_CONT);
-            if (result != CURLE_OK) {
-                DLOGW("Failed to un-pause curl with error: %u. Curl object %p", result, pCurlResponse->pCurl);
-            }
+        if (pCurlResponse->pCurl != NULL) {
+            pCallbacksProvider->clientCallbacks.lockMutexFn(pCallbacksProvider->clientCallbacks.customData, pCurlResponse->dataAvailableLock);

```

Comment:
No need to lock the data lock. You can simply proceed with signaling

Reply:
Yeah, definitely don't need to lock - in this case, the CVAR will not be able to re-acquire the lock until you release it.

Reply:
I basically need to 1.  signal cvar and 2. set the data available boolean to TRUE atomically, hence the lock.  The CVAR is able to re-acquire the lock, I mean this does work, the lock is released immediately after the signal so it should be fine.